magento/magento2
# #18668: Fixed store reinitialization for integration tests on projects with plugins on \Magento\TestFramework\App\Config

### Description (*)
This PR fixes store reinitialization for integration tests done on projects with plugins on `\Magento\Framework\App\Config\ScopeConfigInterface` which effectively makes the system use that plugin on all preferences for that interface.

When you have such a plugin, it forces the system to generate an Interceptor class for the `\Magento\TestFramework\App\Config` class (the implementation of `ScopeConfigInterface` in the integration tests framework). When a reflection is created for that Interceptor class, it doesn't have the 'data' property which its parents has, which results in a `\ReflectionException`  being thrown on `$reflection->getProperty('data')`. In this case, the property has to be fetched from the parent class (the actual Config class).

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#18668

### Manual testing scenarios (*)
Create a module with two things:
1. A plugin for \Magento\Framework\App\Config\ScopeConfigInterface with anything inside, like:
```
public function afterGetValue(
    \Magento\Framework\App\Config\ScopeConfigInterface $subject,
    $result
) {
    return $result;
}
```
di.xml:
```
<type name="Magento\Framework\App\Config\ScopeConfigInterface">
    <plugin name="myplugin" type="MyVendor\MyModule\Plugin\MyPlugin"/>
</type>
```
2. An integration test that creates a new website via fixture or is doing anything else that triggers the `reinitStores()` function. Actual asserts inside the test don't matter, so you can just create an assert that true is true: `$this->assertTrue(true)`

Then run your test. The expected result is that it doesn't fail with "Error in fixture: [path here]. Property data does not exist

### Questions or comments
**IMPORTANT**: Although the issue this PR fixes concerns only projects with customization, not a clean Magento instance, this is exactly the purpose of integration tests framework - to let people test their custom modules (with the only two use cases when it's not testing custom code being the core team and contributors using it to test new code introduced to the core). Because of this I believe it's justified to process this PR with a high priority (higher than P3). Once you hit the issue, there's no workaround for it and you have to patch the integration tests framework to be able to keep using it on your project.

### Contribution checklist (*)
